example_buffer.py

```diff
@@ -0,0 +1,194 @@
+import tensorflow as tf
+import os
+import random
+from tqdm import tqdm
+import multiprocessing as mp
+import functools
+import itertools
+import rl_loop
+import dual_net
+import subprocess
+from utils import timer
+import time
+import preprocessing
+import argh
+import argparse
+import datetime as dt
+
+
+# 1 for ZLIB!  Make this match preprocessing.
+READ_OPTS = tf.python_io.TFRecordOptions(1)
+
+LOCAL_DIR = "data/"
+
+
+def pick_examples_from_tfrecord(filename, samples_per_game=4):
+    protos = [p for p in
+              tf.python_io.tf_record_iterator(filename, READ_OPTS)]
+    if len(protos) < 20:
+        return []
+    choices = random.sample(protos, samples_per_game)
+
+    def make_example(protostring):
+        e = tf.train.Example()
+        e.ParseFromString(protostring)
+        return e
+    examples = list(map(make_example, choices))
+    return examples
+
+
+def choose(g, samples_per_game=4):
+    examples = pick_examples_from_tfrecord(g, samples_per_game)
+    t = file_timestamp(g)
+    return [(t, ex) for ex in examples]
+
+
+def file_timestamp(filename):
+    return int(os.path.basename(filename).split('-')[0])
+
+
+def _ts_to_str(timestamp):
+    return dt.datetime.fromtimestamp(timestamp).strftime("%Y-%m-%d %H:%M:%S")
```

Comment:
isoformat() returns basically the same thing which is close enough for debug string IMHO

Reply:
ack.  Close enough, and i have to do the fromtimestamp() anyway